### ISSUE TITLE
Confusing error message on malformed double literal

### ISSUE THREAD

**Issue by lgolding** [lgolding/auto-scale#15]
When a string like `1.9.0` occurs within a function argument list, for example:

```
a = $ActiveTasks.GetSample(1.9.0)
```

it gives a parse error at the second dot, which is fine. But the error message is

```
Expected a token of one of the types DoubleLiteral, StringLiteral, Identifier,
but got token "." of type OperatorMemberSelect.
```

Why would the parser expect on of those at the point where I typed the second dot?

**Comment by lgolding** [comment]
This was a bug in the parsing of argument lists. After each argument, the next token should be either a `Comma` or a `CloseParen`. If it was neither one, the parser incorrectly went back to get another argument. The error message was due to it not recognizing a dot as the first character of a primary expression.

Now it correctly complains that it was expecting a `Comma` or a `CloseParen`.

**Comment by lgolding** [comment]
Fixed by 2ab40fb.
